src/core/public/chrome/recently_accessed/recently_accessed_service.ts

```diff
@@ -59,15 +61,32 @@ export class RecentlyAccessedService {
 
     return {
       /** Adds a new item to the history. */
-      add: (link: string, label: string, id: string) => {
+      add: (
+        link: string,
+        label: string,
+        id: string,
+        extraProps?: { type?: string; updatedAt?: number }
+      ) => {
         const currentWorkspaceId = workspaces.currentWorkspaceId$.getValue();
-
-        history.add({
-          link,
-          label,
-          id,
-          ...(currentWorkspaceId && { workspaceId: currentWorkspaceId }),
-        });
+        if (extraProps) {
+          const type = extraProps!.type;
+          const updatedAt = extraProps!.updatedAt;
+          history.add({
+            link,
+            label,
+            id,
+            ...(currentWorkspaceId && { workspaceId: currentWorkspaceId }),
+            type,
+            updatedAt,
+          });
+        } else {
+          history.add({
+            link,
+            label,
+            id,
+            ...(currentWorkspaceId && { workspaceId: currentWorkspaceId }),
+          });
+        }
```

Comment:
```suggestion
      history.add({
        link,
        label,
        id,
        ...(currentWorkspaceId && { workspaceId: currentWorkspaceId }),
        ...extraProps,
      });
```
What about we simplifying the code like this?